Refactor Pt. 2

bforcum opened the pull request:
Massive refactoring changes to settings_page, map_page, and panel_page.

bforcum committed:
Refactor widgets and classes out of map_page.dart and panel_page.dart and into subdirectories
Rename hazard_info.dart to hazard_info_card.dart
Make all imports absolute rather than relative

bforcum committed:
Move hazard_info_popup to map_page (misunderstood its purpose last time)
Rename hazard_info_card back to hazard_info (also misunderstood its purpose)

bforcum committed:
Refactor trail_info.dart, comment each sub-page with a brief summary

bforcum committed:
Refactor settings_widgets into more smaller files

elliotnash committed:
Doc wording changes

elliotnash commented on lib/page/home_page/map_page.dart:
```diff
@@ -1,42 +1,41 @@
 
+/// The main app body with all the moving/positioned elements over the main app.
+/// 
+/// Contains all trails, hazard markers, hazard info popups, plus the main map. 
```
I wouldn't say main app page as that implies other UI elements are here. Doc could be as simple as

```dart
/// Renders the main map.
///
/// Contains all trails, hazard markers, and hazard info popups.
```

elliotnash commented on lib/page/home_page/map_page/hazard_marker_layer.dart:
```diff
@@ -0,0 +1,95 @@
+import 'package:forest_park_reports/provider/relation_provider.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+
+/// A layer over the trails with hazard icons and HazardInfoPopup handling
```
When referencing another class in a doc comment, use square brackets to make it link.
```dart
/// ... and [HazardInfoPopup] handling.
```

elliotnash commented on lib/page/home_page/panel_page/platform_pill.dart:
I might move this file to `page/common` since even though it is only used in panel_page for now, we could use it elsewhere in the app in the future (without modification).

elliotnash commented on lib/page/settings_page/setting_confirmation.dart:
```diff
@@ -0,0 +1,37 @@
+        ),
+      ],
+    ),
+  ) as Future<bool>;
```
`showConfirmationDialog` should be async, and then `showPlatformDialog` should be awaited inside the function.

```dart
Future<bool> showConfirmationDialog(...) async {
  return await showPlatformDialog(
    ...
  );
}
```
This likely does not require casting as dart should infer the type parameter of `showPlatformDialog` based on the return type of `showConfirmationDialog`. In the case that it does require casting, you would first await `showPlatformDialog` to resolve the `Future<dynamic>` to just `dynamic`, then cast that to a `bool`.

elliotnash reviewed (changes requested):
A few minor changes

bforcum committed:
PR changes
Move PlatformPill to page/common
Clarify MapPage doc
Format HazardMarkerLayer doc
Make settings showConfirmationDialog async
